crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -662,4 +725,210 @@ mod tests {
         world.flush();
         assert_eq!(1, world.resource::<R>().0);
     }
+
+    #[test]
+    fn observer_propagating() {
```

Comment:
Pedantic: I'm seeing many tests like this, not just for bubbling but observer in general. Tests using counters to determine if multiple observers ran or if a subset of observers ran. This isn't the best testing because the tests can pass when they shouldn't if an observer runs multiple times or if the wrong subset of observers run.

Reply:
simple fix is a `Vec<str>` where we can push observer "labels" when they execute and then verify with `vec!["obs1", "obs2"]`
it's only tests so it doesn't need to be "proper"

would that be alright?

Reply:
That seems correct to me. I'd rather not make this change as part of this PR. Changing the existing tests is out of scope, and I'd like to keep consistency. But I agree these tests could be improved as follow-up work.